source/ui/datasources/ui_blockedplayers_datasource.cpp

```diff
@@ -0,0 +1,44 @@
+#include "ui_precompiled.h"
+#include "kernel/ui_common.h"
+#include "datasources/ui_blockedplayers_datasource.h"
+
+#define TABLE_NAME "blockedplayers"
+
+namespace WSWUI
+{
+    BlockedPlayersDataSource::BlockedPlayersDataSource():Rocket::Controls::DataSource( "blockedplayers" )
+	{
+		const char *blocked = trap::Cvar_String("cg_chatBlocklist");
+		// stored in the format `:name      steamid` (name is padded out to match max)
+
+		for (int i = 0 ;; i += 15+17+1) {
```

Comment:
these magic numbers look arbitrary.

Reply:
steam account's 64 bit id is a 17 digit number.

Reply:
Min: 0x0110000100000000
Max 0x01100001FFFFFFFF
Always = 17 decimal digits.

Reply:
they seem to correspond to the arguments of strncpy below, but yes those should not be as arbitrary numbers scattered around the codebase.